From 15036dd4fa9eb209f5e148c6f7ee081f5ca78fa4 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan at windriver.com>
Date: Fri, 11 Sep 2015 03:37:13 -0400
Subject: [PATCH] audit/auvirt: get inline functions work with C99

This fixes link errors:

  auvirt.c:484: undefined reference to `copy_str'
  auvirt.c:667: undefined reference to `is_resource'

As gcc5 doc about "Different semantics for inline functions":

  > C99 extern inline: An externally visible function is generated; \
  > same as GNU89 inline.

  > To fix this, either mark the function foo as extern, or add the \
  > following declaration: extern inline int foo (void);

(from https://gcc.gnu.org/gcc-5/porting_to.html)

Upstream Status: pending

Signed-off-by: Wenzong Fan <wenzong.fan at windriver.com>
---
 tools/auvirt/auvirt.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/auvirt/auvirt.c b/tools/auvirt/auvirt.c
index 655c454..b16d718 100644
--- a/tools/auvirt/auvirt.c
+++ b/tools/auvirt/auvirt.c
@@ -138,7 +138,7 @@ void event_free(struct event *event)
 	}
 }
 
-inline char *copy_str(const char *str)
+extern inline char *copy_str(const char *str)
 {
 	return (str) ? strdup(str) : NULL;
 }
@@ -650,7 +650,7 @@ int process_control_event(auparse_state_t *au)
 	return 0;
 }
 
-inline int is_resource(const char *res)
+extern inline int is_resource(const char *res)
 {
 	if (res == NULL ||
 	    res[0] == '\0' ||
-- 
1.9.1
